app/jobs/ingest_archival_media_bag_job.rb

```diff
@@ -0,0 +1,227 @@
+# frozen_string_literal: true
+class IngestArchivalMediaBagJob < ApplicationJob
+  BARCODE_WITH_PART_REGEX = /(\d{14}_\d+)_.*/
+
+  def perform(collection_component:, bag_path:, user:)
+    amc = find_or_create_amc(collection_component)
```

Comment:
Do we want to move this into the buffer_into_index part below? Otherwise if something goes wrong we're left with an AMC object created.